xarray/namedarray/core.py

```diff
@@ -0,0 +1,342 @@
+from __future__ import annotations
+
+import copy
+import itertools
+import math
+import typing
+from collections.abc import Hashable, Iterable, Mapping
+
+import numpy as np
+import numpy.typing as npt
+
+from xarray.namedarray.utils import Frozen, _default, is_duck_dask_array
+
+# temporary placeholder for indicating that an array api compliant
+# type. hopefully in the future we can narrow this down more
+T_Array = typing.TypeVar("T_Array", bound=typing.Any)
+
+
+class NamedArray:
```

Comment:
```suggestion
class NamedArray(NamedArrayArithmetic):
```
We do want to be able to do arithemitcs right? We could have a mixin like VariableArithmetic.

Reply:
This could be in a followup with arithmetic ops, and unary ops.